runtime/runtime/src/lib.rs

```diff
@@ -835,22 +858,174 @@ impl Runtime {
                 receipt,
                 &mut new_receipts,
                 &mut validator_proposals,
-                &mut total_rent_paid,
+                &mut stats,
             )?
             .into_iter()
             .for_each(|res| tx_result.push(res));
         }
+
+        self.balance_check(
+            &initial_state,
+            &state_update,
+            prev_receipts,
+            transactions,
+            &new_receipts,
+            &stats,
+        )?;
+
         let trie_changes = state_update.finalize()?;
         Ok(ApplyResult {
             state_root: StateRoot { hash: trie_changes.new_root, num_parts: 9 }, /* TODO MOO */
             trie_changes,
             validator_proposals,
             new_receipts,
             tx_result,
-            total_rent_paid,
+            stats,
         })
     }
 
+    // TODO: Check for balance overflows
+    // TODO: Fix StorageError for partial states when looking up something that doesn't exist.
+    fn balance_check(
+        &self,
+        initial_state: &TrieUpdate,
+        final_state: &TrieUpdate,
+        prev_receipts: &[Receipt],
+        transactions: &[SignedTransaction],
+        new_receipts: &[Receipt],
+        stats: &ApplyStats,
+    ) -> Result<(), InvalidTxErrorOrStorageError> {
+        // Accounts
+        let all_accounts_ids: HashSet<AccountId> = transactions
+            .iter()
+            .map(|tx| tx.transaction.signer_id.clone())
+            .chain(prev_receipts.iter().map(|r| r.receiver_id.clone()))
+            .collect();
+        let total_accounts_balance = |state| -> Result<u128, StorageError> {
+            Ok(all_accounts_ids
+                .iter()
+                .map(|account_id| {
+                    Ok(get_account(state, account_id)?.map_or(0, |a| a.amount + a.locked))
```

Comment:
No. `locked` - is the new name for `staked` tokens, but it's more appropriate, because it doesn't actually reflect the current stake. Because if you desire to unstake, the locked tokens don't get unlocked immediately, but only 3 epochs later.